mochify/index.js

```diff
@@ -18,26 +18,40 @@ async function mochify(options = {}) {
   const mocha_runner = createMochaRunner(config.reporter || 'spec');
   const { mochifyDriver } = resolveMochifyDriver(config.driver);
 
+  const [mocha, client, files] = await Promise.all([
+    readFile(require.resolve('mocha/mocha.js'), 'utf8'),
+    readFile(require.resolve('./client'), 'utf8'),
+    resolveSpec(config.spec)
+  ]);
+
+  const configured_client = setupClient(client, config);
+
   const driver_options = config.driver_options || {};
   let server = null;
-  if (config.serve) {
-    server = await startServer(config.serve, config.server_options);
+  if (config.serve || config.esm) {
+    const scripts = [];
+    if (config.esm) {
+      scripts.push({ content: mocha });
```

Comment:
I wonder if we should always serve them directly, even when users are not using the `esm` option?

Reply:
That's a very good point. Would make the file concatenation go away if no bundler is specified, and give correct stack traces without a source map as well.

Reply:
Although this means there is always a server, no?

Reply:
I think we have 3 options:
- include the "prelude" in the served doc when `--esm` is given (as this requires a server)
- include it in the served doc when `--esm` or `--server` is given (as both require a server)
- always run a server, always include it in the doc

I have to admit I am not entirely sure about all of the consequences choosing option 3 would have so right now my answer would be "i don't know", but I will think about it more.

Reply:
I just realized that option 3 is not a good idea. Drivers that inject scripts into remote computers wouldn't work (e.g. `webdriver`).

Reply:
I changed it so that we use option 2 which I guess makes the most sense.